### ISSUE TITLE
Google Assistant not working with HASS and Cloudflare Proxy mode

### ISSUE THREAD

**Issue by tehniemer** [htpcBeginner/docker-traefik#79]
Google Assistant doesn't seem to be working with cloudflare's proxy, it works if the CNAME entry is set to DNS only though. What do we need to change in cloudflare settings to get the Google Assistant API through the proxy? I followed this [guide](https://www.smarthomebeginner.com/configure-google-assistant-for-home-assistant/).

**Comment by tehniemer** [comment]
These are the errors I see in the logs.
```
2020-08-26 16:22:02 ERROR (MainThread) [homeassistant.components.google_assistant.http] Request for https://homegraph.googleapis.com/v1/devices:reportStateAndNotification failed: 404,
2020-08-26 16:22:11 ERROR (MainThread) [homeassistant.components.google_assistant.http] Could not contact https://homegraph.googleapis.com/v1/devices:reportStateAndNotification
```

**Comment by tehniemer** [comment]
Upon further investigation, I don't think those errors are related to the Cloudflare proxy issue. I disabled proxy for my homeassistant CNAME and I was able to re-connect to google assistant, but those errors are still present in the logs.

**Comment by htpcBeginner** [comment]
It appears you figured this out. So I am closing this.

**Comment by yulasinio** [comment]
Anyone figured how to integrate HA with Google Assistant without exposing the external IP on Cloudflare?
If I modify the CNAME for HA to DNS only the GA integration works but as soon as I switch back to proxy the devices go offline.

Thanks,
Julian

**Comment by marcinprzybysz86** [comment]
nope, moved my domain to Cloudflare to have protection. HASS is on subdomain   ha.example.com and whether i turn on proxying it stops working :(
